### PR TITLE
Add DateTimeZone

### PR DESCRIPTION
**Pull request by chadicus**

### CODE REVIEW

File: src/Filter/DateTimeZone.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * @param boolean $allowNull True to allow nulls through, and false (default) if nulls should not be allowed.
+     *
+     * @return \DateTimeZone
+     *
+     * @throws \InvalidArgumentException Thrown if $allowNull was not a boolean value.
+     * @throws \Exception if the value did not pass validation.
+     */
+    public static function filter($value, $allowNull = false)
+    {
+        if ($allowNull !== false && $allowNull !== true) {
```
Comment (chrisryan) [inline src/Filter/DateTimeZone.php]: curious why you use this format rather than `if (!is_bool($allowNull)) {`?

**Comment by chrisryan** [comment]
I would add a test for non-string value.

```php
    /**
     * Verify behavior of filter() when $value is not a string.
     *
     * @test
     * @covers ::filter
     * @expectedException \InvalidArgumentException
     * @expectedExceptionMessage $value not a non-empty string
     */
    public function filter_nonStringArgument()
    {
        $timezone = TZ::filter(42);
    }
```

File: src/Filter/DateTimeZone.php (hunk 1 above)
Comment (chadicus) [inline src/Filter/DateTimeZone.php]: we found out a while back that `!== false && !== true` is actually faster than `is_bool`

Comment (chrisryan) [inline src/Filter/DateTimeZone.php]: good to know

**Comment by chrisryan** [comment]
:+1:

**Pull request merged**